src/fun/counting_v2.go.tmpl

```diff
@@ -0,0 +1,268 @@
+			"description" (printf "```Pos    ✅  User%s```" $desc) 
+			"footer" (sdict "text" "Use this command: -CStats Leaderboard") 
+			"color" 30654
+		)}}
+
+	{{else}}
+		{{sendMessage nil (cembed 
+			"title" "Invalid Syntax" 
+			"description" "For server counting statistics: `-CStats`\nFor a members' statistics: `-CStats <User: @/ID>`\nFor server leaderboard: `-CStats Leaderboard`" 
+			"color" 16744192
+		)}}
+	{{end}}
+	{{return}}
+{{end}}
+
+{{if and (eq $db.last.user .User.ID) (not $countTwice)}}
+	{{sendMessage nil (cembed 
+		"description" (printf "You can't count twice in a row 🥲\nThe next number is %d" $db.next) 
+		"color" 16744192
+	)}}
+	{{return}}
+{{end}}
+
+{{$invalid:=false}}
+{{$re:=reFindAll `\+|\-|\*|\/|\^|\%|a?sin|a?cos|a?tan|a?cot|a?sec|a?csc|sqrt|l(o?g|n)|abs|e|pi|π|\(|\)|\d+|M{0,3}(CM|CD|D?C{0,3})(XC|XL|L?X{0,3})(IX|IV|V?I{0,3})` .Cmd}}
+{{if ne .Cmd (joinStr "" $re)}}
+	{{$invalid =true}}
+{{end}}
+
+{{$convertRN:=sdict "I" 1 "V" 5 "X" 10 "L" 50 "C" 100 "D" 500 "M" 1000}}
+{{$exp:=""}}{{$lastNumeral:=false}}
+{{range $re}}
+	{{- $roman:=reFindAll `I|V|X|L|C|D|M` .}}
+	{{- if eq (len $roman) 0}}
+		{{- $exp =printf "%s%s" $exp .}}
+		{{- if reFind `\d+` .}}
+			{{- if $lastNumeral}}
+				{{$invalid =true}}{{- break}}
+			{{- end}}
+			{{- $lastNumeral =true}}
+		{{- else}}
+			{{- $lastNumeral =false}}
+		{{- end}}
+		{{- continue}}
+	{{- end}}
+	{{- if $lastNumeral}}
+		{{$invalid =true}}{{- break}}
+	{{- end}}
+	{{- $values:=cslice}}{{- $value:=0}}{{- $skip:=false}}
+	{{- range $roman}}
+		{{- $values =$values.Append ($convertRN.Get .)}}
+	{{- end}}
+	{{- range $i,$v:=$values}}
+		{{- $i =add $i 1}}
+		{{- if $skip}}
+			{{- $skip =false}}
+			{{- continue}}
+		{{- end}}
+		{{- if le (len $values) $i}}
+			{{- $value =add $value .}}
+			{{- continue}}
+		{{- end}}
+		{{- if ge . (index $values $i)}}
+			{{- $value =add $value .}}
+		{{- else}}
+			{{- $value =sub (add $value (index $values $i)) .}}
+			{{- $skip =true}}
+		{{- end}}
+	{{- end}}
+	{{- $exp =printf "%s%d" $exp $value}}
+	{{- $lastNumeral =true}}
+{{end}}
+
+{{$number:=0}}
+{{try}}{{$number =slice ($number =exec "calc" $exp) 9 (len $number|add -1)|toFloat|toInt}}
+{{catch}}{{$invalid =true}}{{end}}
+
+{{if $invalid}}
+	{{sendMessage nil (cembed 
+		"description" (printf "`%s` sent an invalid expression, `%s`\nNext number is `%d`" .User.Username .Cmd $db.next) 
+		"color" 16711680
+	)}}
+	{{return}}
+{{end}}
+
+{{$errFormat:="Counting Error: `%s`\nMessage: %s\nUser: `%s` (%d)"}}
+
+{{if eq $db.next $number}} {{/* correct */}}
+	{{$db.Set "next" (add $db.next 1)}}
+	{{if $reactions}}
+		{{try}}
+			{{addReactions $correctEmoji}}
+			{{if and $reactionDelete $db.last.msg}}
+				{{deleteMessageReaction nil $db.last.msg .BotUser.ID $correctEmoji}}
+			{{end}}
+			{{if eq (mod $number 100|toInt) 0}}{{addReactions "💯"}}{{end}}
+		{{catch}}
+			{{with getRole $incorrectRID}}
+				{{addRoleID .ID}}{{(toDuration "1d").Seconds|toInt|removeRoleID .ID}}
+			{{end}}
```

Comment:
You can use `addRole` with the role object directly, and likewise with `removeRole`.

Reply:
Out of curiosity, what benefit does this have; just one less step to get the ID value?

Reply:
I assumed the length of the code was a minor concern given the compressed style, and `addRole .` is shorter without compromising too much on clarity.